name: Validate URLs

on:
  workflow_dispatch:
  schedule:
    - cron: '0 13 * * 1'

jobs:

  validate_urls:
    name: Validate URLs
    runs-on: ubuntu-latest
    environment: 'production'
    permissions:
      contents: read
      issues: write
    steps:
      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2

      - name: Setup Python
        uses: actions/setup-python@a26af69be951a213d495a4c3e4e4022e16d87065 # v5.6.0
        with:
          python-version: 3.9
          architecture: x64

      - uses: astral-sh/setup-uv@bd01e18f51369d5a26f1651c3cb451d3417e3bba # v6.3.1
        with:
          version: ~=0.5.27

      - name: Get invalid URLs
        id: urls
        run: |
          output=$(uv run python utility_scripts/plugin_definitions/check_links.py)
          echo "INVALID_URL<<EOF" >> $GITHUB_ENV
          echo "$output" >> $GITHUB_ENV
          echo "EOF" >> $GITHUB_ENV
      - uses: JasonEtco/create-an-issue@1b14a70e4d8dc185e5cc76d3bec9eab20257b2c5 # v2.9.2
        if: env.INVALID_URL != ''
        env:
          GITHUB_TOKEN: ${{ secrets.MELTYBOT_GITHUB_AUTH_TOKEN }}
          INVALID_URLS: ${{ env.INVALID_URL }}
        with:
          filename: .github/ISSUE_TEMPLATE/invalid_urls.md
          update_existing: true
